rpmlint/checks/ProductCheck.py

```diff
@@ -0,0 +1,46 @@
+import re
+import stat
+import subprocess
+
+from xml.dom.minidom import parse
+from urllib.parse import unquote
+
+from rpmlint.checks.AbstractCheck import AbstractFilesCheck
+
+
+class ProductCheck(AbstractFilesCheck):
+    """
+    Validate that product files are correct. currently only cpeid.
+    """
+
+    def __init__(self, config, output):
+        super().__init__(config, output, r'/etc/products.d/.*\.prod$')
+
+    def check_file(self, pkg, filename):
+        cpeid_provider_found = None
+        cpeid_xml_found = None
+        try:
+            for provide in pkg.provides:
+                if provide.name == 'product-cpeid()' and len(provide.version) > 1:
+                      cpeid_provider_found = unquote(provide.version[1])
+
+            if not cpeid_provider_found:
+                self.output.add_info('E', pkg, 'product-parsing-exception', 'no product-cpeid() provider', filename)
+
+            lf = pkg.dir_name() + filename
+            xml = parse(lf)
+            for cpeid in xml.getElementsByTagName('cpeid'):
+                cpeid_xml_found = cpeid.firstChild.data
+
+            if not cpeid_xml_found:
+                self.output.add_info('E', pkg, 'product-parsing-exception', 'no cpeid defined in prod file', lf)
+
+            if cpeid_xml_found != cpeid_provider_found:
+                self.output.add_info('E', pkg, 'product-parsing-exception', 'cpeid defined different in prod file to rpm provides', lf)
+
+        except Exception as e:
```

Comment:
never create a "general exception" handling across many, can we just have it where it is actually necessary?